### ISSUE TITLE
False positive in PYI files with ellipsis for `PIE796` 

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#8818]
If I define an enum file with ellipsis in PYI files, the linter complains
```
class CreationMeta(Enum):
    DEFAULT = ...
    IN_CUSTOM_FUNCTION = ...
    MULTI_OUTPUT_NODE = ...
    NO_GRAD_MODE = ...
    INFERENCE_MODE = ...
```

in a .PYI files, these values are all set to `...` to denote that they will be populated in other code, and we only care about what the members of the enum are. This is a pretty common idiom so we may want to explicitly whitelist in our check or at least propose a better fix for PYI files as I am unaware of another way of doing this with a placeholder value for `Enum`.

**Comment by charliermarsh** [comment]
Agree, this makes sense to fix IMO.

**Comment by Avasam** [comment]
> I am unaware of another way of doing this with a placeholder value for `Enum`.

With pyright in basic mode, or with mypy the following is equivalent
```
class CreationMeta(Enum):
    DEFAULT: Any
    IN_CUSTOM_FUNCTION: Any
    MULTI_OUTPUT_NODE: Any
    NO_GRAD_MODE: Any
    INFERENCE_MODE: Any
```
```
from _typeshed import Incomplete
class CreationMeta(Enum):
    DEFAULT: Incomplete
    IN_CUSTOM_FUNCTION: Incomplete
    MULTI_OUTPUT_NODE: Incomplete
    NO_GRAD_MODE: Incomplete
    INFERENCE_MODE: Incomplete
```

If they are indeed classvars:
```
from typing import ClassVar
class CreationMeta(Enum):
    DEFAULT: ClassVar
    IN_CUSTOM_FUNCTION: ClassVar
    MULTI_OUTPUT_NODE: ClassVar
    NO_GRAD_MODE: ClassVar
    INFERENCE_MODE: ClassVar
```

Note that using Any/Incomplete with pyright in strict mode also works, it just changes the underlying type (from `Unknown` to `Any`). but if you're using pyright in strict mode, you really ought to specify your type anyway.

**Comment by Avasam** [comment]
In a statically type-checked context, I feel this would still be undesirable, but that's more for a rule like https://github.com/PyCQA/flake8-pyi/issues/98
